### PR TITLE
[4.0] Repair add new user note

### PR DESCRIPTION
**Pull request by alexandraciobica**
### Summary of Changes
- Added the missing return statement to check() in NoteTable.
- Added default 0 to modified_user_id in user_notes table.

Now the user note is created.

### Steps to reproduce the issue
1. Login to backend
2. Go to User Notes > New
3. Fill in the required data > Save

### Expected result
New user note is created.

### Actual result
An error occurs when trying to save.
screen shot 2018-06-22 at 12 21 20

### Additional comments

The check method from NotesTable misses the return statement.
If this is fixed, the following error occurs when trying to save:
screen shot 2018-06-22 at 12 25 46

### CODE REVIEW

**Comment by ghost** [comment]
Changed Title to show its about 4.0.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20873">issues.joomla.org/tracker/joomla-cms/20873</a>.</sub>

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE `#__user_notes`
+	CHANGE COLUMN `modified_user_id` `modified_user_id` int(10) unsigned NOT NULL DEFAULT 0;
```
Comment (laoneo) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: New line is missing.

File: administrator/components/com_admin/sql/updates/postgresql/4.0.0-2018-06-26.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE "#__user_notes"
+	ALTER COLUMN "modified_user_id" SET DEFAULT 0;
```
Comment (laoneo) [inline administrator/components/com_admin/sql/updates/postgresql/4.0.0-2018-06-26.sql]: New line is missing.

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql (hunk 1 above)
Comment (alikon) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: you don't need to specify NOT NULL  as it already declared not null
``modified_user_id` int(10) unsigned NOT NULL``

Comment (alexandraciobica) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: I tested it without NOT NULL and it makes the modified_uder_id accept nulls.

Comment (alikon) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: see https://github.com/joomla/joomla-cms/blob/staging/installation/sql/mysql/joomla.sql#L223

Comment (alikon) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: see https://github.com/joomla/joomla-cms/blob/staging/installation/sql/mysql/joomla.sql#L2048

Comment (wilsonge) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: This is the same as what's in this file. Not sure exactly why you're linking there @alikon ?

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE `#__user_notes` CHANGE COLUMN `modified_user_id` `modified_user_id` int(10) unsigned NOT NULL DEFAULT 0;
```
Comment (wilsonge) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: This needs to match the schema parser expected format https://github.com/joomla/joomla-cms/blob/6de2b00858e91bd64e7c410a6ed617d455f62d64/libraries/src/Schema/ChangeItem/MysqlChangeItem.php#L167-L187 - doesn't look like it does right now

Comment (alexandraciobica) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: Fixed it. But then, I think this one is also wrong https://github.com/joomla/joomla-cms/blob/9c86ac8482203115608bac8920badcadf23fa322/administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-04-14.sql#L29-L45
It doesn't relate to this pr, I just looked into it when writing mine.

Comment (laoneo) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: Then please open a new pr with the fixes. Thanks for spotting.

Comment (wilsonge) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-26.sql]: That is definitely wrong - a PR fix for that would be good!

**Comment by wilsonge** [comment]
Looks good to me! Thanks!

**Pull request merged**